saleor/product/migrations/0111_auto_20191209_0349.py

```diff
@@ -0,0 +1,78 @@
+# Generated by Django 2.2.7 on 2019-12-09 09:49
+from django.db import migrations
+from django.db.models import Count, Sum
+
+
+def get_or_create_warehouse(apps):
+    Warehouse = apps.get_model("warehouse", "Warehouse")
```

Comment:
I think that this logic should be as a second step inside a `warehouse.init_migration`

Reply:
In the beginning, I was thinking the same. But I don't want to force-create someone object in DB if I don't need it. Besides - someone can delete it between migrations.

Reply:
I am not sure that I understand it correctly. `But I don't want to force-create someone object in DB if I don't need it.` -> the result will be the same for both approaches.
`Besides - someone can delete it between migrations.` -> We could solve the proble by using `dependencies` between given migrations

Reply:
`the result will be the same for both approaches.` - not exactly. In your approach, we will create a warehouse no matter if we really need it or not. In my approach - we will create warehouse if a) there are some product variants created and b) there is no other warehouse that supports all shipping zones.

dependencies do not guarantee that created warehouse will still exist in time for migrating stocks.